### PR TITLE
Using existing shared podLister instead

### PR DESCRIPTION
**Pull request by PlatformLC**
**Please provide a description of this PR:**

#42372 follow-up: Using existing shared podLister instead

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [x] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: cni/pkg/ambient/net.go [hunk 1]
```diff
@@ -238,20 +240,15 @@ func buildEbpfArgsByIP(ip string, isZtunnel, isRemove bool) (*ebpf.RedirectArgs,
...
 	// We assume per node POD's CIDR is the same block, so the route to the POD
 	// from host should be "same". Otherwise, there may multiple host IPs will be
 	// used as source to dial to PODs.
-	pods, err := kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(
-		context.TODO(),
-		metav1.ListOptions{
-			LabelSelector: "app=ztunnel",
-			FieldSelector: "spec.nodeName=" + NodeName,
-		})
+	pods, err := podLister.List(labels.Set{"app": "ztunnel"}.AsSelector())
```
Comment (hzxuzhonghu) [inline cni/pkg/ambient/net.go:247]: do not equal

Comment (howardjohn) [reply cni/pkg/ambient/net.go:247]: Yes, don't we need the nodeName selector?

Comment (PlatformLC) [reply cni/pkg/ambient/net.go:247]: Initially, I think while this piece of code called, we should have an active ztunnel, thus stick to ztunnel pod here.

Comment (PlatformLC) [reply cni/pkg/ambient/net.go:247]: this lister should share the podInformer's  Indexer, which has already initiated with the nodeName selector. refer to

[podInformer](https://github.com/istio/istio/blob/af83bc1e0fdffe6bbd3317de81de36e33d53a9e4/cni/pkg/ambient/informers.go#L48)

Comment (howardjohn) [reply cni/pkg/ambient/net.go:247]: nice

**Review by howardjohn — APPROVED** [review]

**Pull request merged**